File: modules/agent/src/models/Partnership.py
from sqlalchemy import Column, Text, BigInteger, Date, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship
from models import Base


class Partnership(Base):
    """Partnership SQLAlchemy model (extends Lead)."""

    __tablename__ = "Partnership"

    id = Column(BigInteger, ForeignKey("Lead.id", ondelete="CASCADE"), primary_key=True)
    partnership_type = Column(Text, nullable=True)
    partnership_name = Column(Text, nullable=False)
    start_date = Column(Date, nullable=True)
    end_date = Column(Date, nullable=True)
    description = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    # Relationships
    lead = relationship("Lead", back_populates="partnership")

File: modules/agent-go/internal/models/Partnership.go
package models

import "time"

type Partnership struct {
	ID              int64      `gorm:"primaryKey" json:"id"` // FK to Lead.id
	OrganizationID  int64      `gorm:"not null" json:"organization_id"`
	PartnershipType *string    `json:"partnership_type"`
	PartnershipName string     `gorm:"not null" json:"partnership_name"`
	StartDate       *time.Time `gorm:"type:date" json:"start_date"`
	EndDate         *time.Time `gorm:"type:date" json:"end_date"`
	Notes           *string    `json:"notes"`
	CreatedAt       time.Time  `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt       time.Time  `gorm:"autoUpdateTime" json:"updated_at"`
}

func (Partnership) TableName() string {
	return "Partnership"
}
